test/rules/no-async-without-await/test.ts.lint

```diff
@@ -0,0 +1,85 @@
+async function a(){
```

Comment:
A couple missing classes of test cases:
* Function generators: `function* foo() { ... }`.
* `async` class ` = ` members _(this just has the non-async version)_